modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -6,21 +6,27 @@ import cats.syntax.applicativeError._
 import cats.syntax.either._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
+import cats.syntax.option._
 
 import org.tessellation.kryo.KryoSerializer
-import org.tessellation.rosetta.domain._
-import org.tessellation.rosetta.domain.error.{ConstructionError, InvalidPublicKey, MalformedTransaction}
-import org.tessellation.rosetta.domain.operation.Operation
+import org.tessellation.rosetta.domain.{AccountIdentifier, RosettaPublicKey, TransactionIdentifier}
+import org.tessellation.rosetta.domain.amount.{Amount, AmountValue}
+import org.tessellation.rosetta.domain.currency.DAGCurrency
+import org.tessellation.rosetta.domain.error.{ConstructionError, DeserializationFailure, InvalidPublicKey, MalformedTransaction}
+import org.tessellation.rosetta.domain.operation.{Operation, OperationIdentifier, OperationIndex, OperationType}
 import org.tessellation.schema.transaction.Transaction
+import org.tessellation.schema.address.Address
 import org.tessellation.security.SecurityProvider
 import org.tessellation.security.hex.Hex
 import org.tessellation.security.key.ops._
 import org.tessellation.security.signature.Signed
+import eu.timepit.refined.auto._
 
 trait ConstructionService[F[_]] {
   def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier]
   def getAccountIdentifiers(operations: List[Operation]): Option[NonEmptyList[AccountIdentifier]]
   def getTransactionIdentifier(hex: Hex): EitherT[F, ConstructionError, TransactionIdentifier]
+  def parseSignedTransaction(hex: Hex): EitherT[F, ConstructionError, (Option[NonEmptyList[Operation]], Option[NonEmptyList[Address]], Option[NonEmptyList[AccountIdentifier]])]
```

Comment:
I don't like this return type: `EitherT[F, ConstructionError, (Option[NonEmptyList[Operation]], Option[NonEmptyList[Address]], Option[NonEmptyList[AccountIdentifier]])]` and recommendations are appreciated.